cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -100,9 +100,17 @@ func stampedAction(stamp string, action func() error) error {
 	return ioutil.WriteFile(stampFile, nil, 0644)
 }
 
-func generateInitramfsMounts() error {
+func generateInitramfsMounts() (err error) {
+	// ensure that the last thing we do is to lock access to sealed keys,
+	// regardless of mode or early failures.
+	defer func() {
+		if e := secbootLockTPMSealedKeys(); e != nil {
+			err = fmt.Errorf("error locking access to sealed keys: %v", e)
```

Comment:
If "generateInitramfsMoutns()" has "err" already set, this will overwrite the pre-exiting error, no? If so I think that is problematic.

Reply:
yes, I think if err is already set we should log this problem but preserve the original error

Reply:
@chrisccoulson @mvo5 I'm taking care of this